XamarinCommunityToolkit/ObjectModel/AsyncCommand.shared.cs

```diff
@@ -0,0 +1,155 @@
+
+		/// <summary>
+		/// Create a new AsyncCommand
+		/// </summary>
+		/// <param name="execute">The Function executed when Execute or ExecuteAsync is called. This does not check canExecute before executing and will execute even if canExecute is false</param>
+		/// <param name="canExecute">The Function that verifies whether or not AsyncCommand should execute.</param>
+		/// <param name="onException">If an exception is thrown in the Task, <c>onException</c> will execute. If onException is null, the exception will be re-thrown</param>
+		/// <param name="continueOnCapturedContext">If set to <c>true</c> continue on captured context; this will ensure that the Synchronization Context returns to the calling thread. If set to <c>false</c> continue on a different context; this will allow the Synchronization Context to continue on a different thread</param>
+		public AsyncCommand(
+			Func<T, Task> execute,
+			Func<T, bool> canExecute = null,
+			Action<Exception> onException = null,
+			bool continueOnCapturedContext = false)
+		{
+			this.execute = execute ?? throw new ArgumentNullException(nameof(execute));
+			this.canExecute = canExecute;
+			this.onException = onException;
+			this.continueOnCapturedContext = continueOnCapturedContext;
+		}
+
+		/// <summary>
+		/// Invoke the CanExecute method and return if it can be executed.
+		/// </summary>
+		/// <param name="parameter">Parameter to pass to CanExecute.</param>
+		/// <returns>If it can be executed.</returns>
+		public bool CanExecute(object parameter)
+		{
+			if (!IsValidParameter(parameter))
+				return false;
+			if (IsExecuting && !AllowMultipleExecution)
+				return false;
+
+			return canExecute?.Invoke((T) parameter) ?? true;
+		}
+
+		/// <summary>
+		/// Event triggered when Can Execute changes.
+		/// </summary>
+		public event EventHandler CanExecuteChanged
+		{
+			add => weakEventManager.AddEventHandler(value);
+			remove => weakEventManager.RemoveEventHandler(value);
+		}
+
+		public async void Execute(object parameter) => await ExecuteAsync((T) parameter);
+
+		/// <summary>
+		/// Execute the command async.
+		/// </summary>
+		/// <returns>Task of action being executed that can be awaited.</returns>
+		public async Task ExecuteAsync(T parameter)
+		{
+			try
+			{
+				IsExecuting = Interlocked.Increment(ref executionCount) > 0;
+				await execute(parameter).ConfigureAwait(continueOnCapturedContext);
+			}
+			catch (Exception e) when (onException != null)
+			{
+				onException(e);
+			}
+			finally
+			{
+				IsExecuting = Interlocked.Decrement(ref executionCount) > 0;
+			}
+		}
+
+		/// <summary>
+		/// Raise a CanExecute change event.
+		/// </summary>
+		public void RaiseCanExecuteChanged() =>
+			weakEventManager.HandleEvent(this, EventArgs.Empty, nameof(CanExecuteChanged));
+
+		static bool IsValidParameter(object o)
+		{
+			if (o != null)
+			{
+				// The parameter isn't null, so we don't have to worry whether null is a valid option
+				return o is T;
+			}
```

Comment:
please don't use braces for one line if-else bodies :)

Just :
if (o != null)
    return o is T;